src/gluonts/dataset/loader.py

```diff
@@ -135,6 +132,9 @@ class TrainDataLoader(DataLoader):
         Number of batches to return in one complete iteration over this object.
     float_type
         Floating point type to use.
+    use_training_transform
+        Enables training mode for transformations, i.e. with the flag
+    `is_train = True`.
```

Comment:
Why is this required? It seems unused in what follows

Reply:
I've missed a place where is should have been located: `model/estimator.py`, in the constructor for validation data loader. We need to perform the correct transformation on validation, so that's why this option is here.
What do you think about the name for this DataLoader. Should it be changed due to the fact that it is also used for validation?

Reply:
I think this can lead to confusion, because of how things are currently. As of now, the `InferenceDataLoader` is available that has this flag hardcoded to `False`. What this implies is that iterating through the `InferenceDataLoader` yields batches sliced from the trailing part of all time series (in particular, having the "prediction" interval at the very end), and each time series is considered only once: this is used to construct the "test" data loader, but I guess it should be the same for the validation one. This way a user can simply slice the train/validation/test dataset appropriately, e.g. (suppose prediction length is `-----`)

```
train:        +---------------------------------------+
validation:   +---------------------------------------|-----+
test:         +---------------------------------------------|-----+
```

and the corresponding data loaders will yield batches with the prediction interval lying in the right place (not overlapping with training data, nor with validation in the test case).

Does this make any sense? In summary, I would remove this option here, and set the `validation_data_loader` to be `InferenceDataLoader` in case validation data is provided.